go.mod

```diff
@@ -76,4 +76,5 @@ require (
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
 
+// need update?
 replace github.com/vmware-tanzu/velero => github.com/openshift/velero v0.10.2-0.20231024175012-d8101a298016
```

Comment:
This seems wrong imo. First of all the upstream version of that dependency is pointing always to tag 0.0.0 and we somehow are using not only hashes but also tags:

https://github.com/vmware-tanzu/velero/blob/c78fea32042c71939f75dc4cc80a9aab92099daf/go.mod#L180

Secondly tag v0.10.2 is not even in the openshift/velero I could find v0.10.1, but not 0.10.2.

I started writing small shell script (not tested yet, but wondering what is your opinion about that way of updating this dependency):


```shell
#!/bin/bash

UPSTREAM_VELERO="github.com/vmware-tanzu/velero"
OPENSHIFT_VELERO="github.com/openshift/velero"

for cmd in go git sed; do
    if ! command -v $cmd &> /dev/null; then
        echo "ERROR: '$cmd' command is not available. Please install $cmd and try again."
        exit 1
    fi
done

cleanup() {
    if [[ -d "$TEMP_DIR" ]]; then
        rm -rf "$TEMP_DIR"
    fi
}

trap cleanup EXIT

TEMP_DIR=$(mktemp -d)

if [[ ! "$TEMP_DIR" || ! -d "$TEMP_DIR" ]]; then
    echo "Could not create temporary directory"
    exit 1
fi

git clone https://$OPENSHIFT_VELERO.git "$TEMP_DIR"
if [[ $? -ne 0 ]]; then
    echo "Failed to clone repository $OPENSHIFT_VELERO"
    exit 1
fi

pushd "$TEMP_DIR" || { echo "Failed to change directory to $TEMP_DIR"; exit 1; }

BASE_VERSION="v0.0.0"

# go requires 12 chars canonical version
LATEST_COMMIT=$(git rev-parse --short=12 HEAD)
if [[ $? -ne 0 || -z "$LATEST_COMMIT" ]]; then
    echo "Failed to retrieve latest commit hash"
    exit 1
fi

LATEST_COMMIT_DATE=$(git show -s --format=%cI $LATEST_COMMIT | sed 's/[-:]//g' | cut -d+ -f1 | cut -dT -f1,2)
if [[ $? -ne 0 ]]; then
    echo "Failed to get current date"
    exit 1
fi

NEW_VERSION="$BASE_VERSION-0.$LATEST_COMMIT_DATE-$LATEST_COMMIT"
popd || { echo "Failed to change back to the original directory"; exit 1; }

sed -i.bak "s|$UPSTREAM_VELERO => $OPENSHIFT_VELERO .*|$UPSTREAM_VELERO => $OPENSHIFT_VELERO $NEW_VERSION|" go.mod
if [[ $? -ne 0 ]]; then
    echo "Failed to update go.mod file"
    exit 1
fi

rm go.mod.bak

go mod tidy
if [[ $? -ne 0 ]]; then
    echo "Failed to tidy go.mod and go.sum files"
    exit 1
fi

git add go.mod go.sum
git commit -m "Update $OPENSHIFT_VELERO to the latest commit $LATEST_COMMIT"
if [[ $? -ne 0 ]]; then
    echo "Failed to commit changes"
    exit 1
fi

echo "Dependency successfully updated to $NEW_VERSION"
```

Reply:
I really like the idea of having an automated check if our dependencies need updates

BUT NOT written in shell (Go, Python, etc, something that is easy to maintain later)

Reply:
Most of hack and surrounding tools are in fact shell based, see:

https://github.com/vmware-tanzu/velero/tree/main/hack

I am not opposed to your idea, but python isn't really feasible in this project as I don't want to introduce yet another language dependency in here.

Go could be an option, but it will be wrapper around native CLIs commads like `git` and will use os `exec.Command` so in fact will introduce more complexity to the simple hack script imo and will not give us benefit of being cross platform. Shell uses them naively without such wrappers and handing of signals/exit values.

Reply:
if if the script did something simpler, like remove go.sum and re-run go mod tidy, if there is a change, then you fail the test and make the user deal with it?

Reply:
lets move this discussion to an issue

Reply:
created issue https://github.com/migtools/oadp-non-admin/issues/76
closing conversation here so we can focus on the PR